packages/amplify-graphql-model-transformer/src/resources/model-resource-generator.ts

```diff
@@ -0,0 +1,670 @@
+import {
+  DataSourceProvider, MutationFieldType, QueryFieldType, SubscriptionFieldType,
+  TransformerContextProvider,
+  TransformerResolverProvider,
+} from '@aws-amplify/graphql-transformer-interfaces';
+import { ObjectTypeDefinitionNode } from 'graphql';
+import { MappingTemplate } from '@aws-amplify/graphql-transformer-core';
+import {
+  ResolverResourceIDs,
+  toCamelCase,
+} from 'graphql-transformer-common';
+import {
+  generateAuthExpressionForSandboxMode,
+  generateResolverKey,
+  ModelVTLGenerator,
+} from '../resolvers';
+import { ModelDirectiveConfiguration, SubscriptionLevel } from '../directive';
+import { ModelTransformerOptions } from '../types';
+
+/**
+ * Abstract class definition for ModelResourceGenerator
+ * ModelResourceGenerator implementations are intended to create resources for the model transformer plugin
+ * according to the data source that backs the model
+ */
+export abstract class ModelResourceGenerator {
+  protected datasourceMap: Record<string, DataSourceProvider> = {};
+  private resolverMap: Record<string, TransformerResolverProvider> = {};
```

Comment:
Moving these to DDB and RDS specific instances make it harder to find collisions?

Reply:
Which collisions are we looking for? The schema is parsed as a whole which prevents model collisions, and the resolver manager used in the transformer context is already blocking resolver collisions

Reply:
Possibly we can create these instances outside resource generator and inject the same instance to both the generators.